app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class CoursesController < ApplicationController
+  def show
+    @course = Course.find params[:id]
```

Comment:
you should use `includes` for group, subject and teacher against `find_records` method
http://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations

and then refactor show view with these changes